diff --git a/node_modules/@greenwood/plugin-css-modules/src/types/index.d.ts b/node_modules/@greenwood/plugin-css-modules/src/types/index.d.ts
index a28b5e8..4b4fcf7 100644
--- a/node_modules/@greenwood/plugin-css-modules/src/types/index.d.ts
+++ b/node_modules/@greenwood/plugin-css-modules/src/types/index.d.ts
@@ -1,6 +1,6 @@
-import type { Plugin } from "@greenwood/cli";
+import type { ResourcePlugin } from "@greenwood/cli";
 
-export type CssModulesPlugin = () => Array<Plugin>;
+export type CssModulesPlugin = () => ResourcePlugin;
 
 declare module "@greenwood/plugin-css-modules" {
   export const greenwoodPluginCssModules: CssModulesPlugin;
