diff --git a/node_modules/@greenwood/plugin-adapter-vercel/package.json b/node_modules/@greenwood/plugin-adapter-vercel/package.json
index 9a9fabc..6dcadc5 100644
--- a/node_modules/@greenwood/plugin-adapter-vercel/package.json
+++ b/node_modules/@greenwood/plugin-adapter-vercel/package.json
@@ -16,8 +16,13 @@
     "Vercel",
     "Edge"
   ],
-  "main": "src/index.js",
   "type": "module",
+  "exports": {
+    ".": {
+      "types": "./src/types/index.d.ts",
+      "import": "./src/index.js"
+    }
+  },
   "files": [
     "src/"
   ],
diff --git a/node_modules/@greenwood/plugin-adapter-vercel/src/index.js b/node_modules/@greenwood/plugin-adapter-vercel/src/index.js
index 821d3d4..c08231f 100644
--- a/node_modules/@greenwood/plugin-adapter-vercel/src/index.js
+++ b/node_modules/@greenwood/plugin-adapter-vercel/src/index.js
@@ -140,7 +140,8 @@ async function vercelAdapter(compilation, options) {
   });
 }
 
-const greenwoodPluginAdapterVercel = (options = {}) => [
+/** @type {import('./types/index.d.ts').VercelAdapter} */
+export const greenwoodPluginAdapterVercel = (options = {}) => [
   {
     type: "adapter",
     name: "plugin-adapter-vercel",
@@ -151,5 +152,3 @@ const greenwoodPluginAdapterVercel = (options = {}) => [
     },
   },
 ];
-
-export { greenwoodPluginAdapterVercel };
diff --git a/node_modules/@greenwood/plugin-adapter-vercel/src/types/index.d.ts b/node_modules/@greenwood/plugin-adapter-vercel/src/types/index.d.ts
new file mode 100644
index 0000000..5b641a6
--- /dev/null
+++ b/node_modules/@greenwood/plugin-adapter-vercel/src/types/index.d.ts
@@ -0,0 +1,13 @@
+import type { Plugin } from "@greenwood/cli";
+
+type SUPPORTED_NODE_VERSIONS = "nodejs22.x" | "nodejs20.x" | "nodejs18.x";
+
+type VercelAdapterOptions = {
+  runtime?: SUPPORTED_NODE_VERSIONS;
+};
+
+export type VercelAdapter = (options?: VercelAdapterOptions) => Array<Plugin>;
+
+declare module "@greenwood/plugin-adapter-vercel" {
+  export const greenwoodPluginAdapterVercel: VercelAdapter;
+}
\ No newline at end of file
