diff --git a/node_modules/@greenwood/plugin-import-raw/src/types/index.d.ts b/node_modules/@greenwood/plugin-import-raw/src/types/index.d.ts
index 5f5384e..0f471ee 100644
--- a/node_modules/@greenwood/plugin-import-raw/src/types/index.d.ts
+++ b/node_modules/@greenwood/plugin-import-raw/src/types/index.d.ts
@@ -1,4 +1,4 @@
-import type { Plugin } from "@greenwood/cli";
+import type { ResourcePlugin } from "@greenwood/cli";
 
 type SUPPORTED_NODE_VERSIONS = "nodejs22.x" | "nodejs20.x" | "nodejs18.x";
 
@@ -6,7 +6,7 @@ type ImportRawPluginOptions = {
   matches?: string[];
 };
 
-export type ImportRawPlugin = (options?: ImportRawPluginOptions) => Array<Plugin>;
+export type ImportRawPlugin = (options?: ImportRawPluginOptions) => ResourcePlugin;
 
 declare module "@greenwood/plugin-import-raw" {
   export const greenwoodPluginImportRaw: ImportRawPlugin;
