diff --git a/node_modules/@greenwood/cli/src/types/index.d.ts b/node_modules/@greenwood/cli/src/types/index.d.ts
index 886e12a..8a37b31 100644
--- a/node_modules/@greenwood/cli/src/types/index.d.ts
+++ b/node_modules/@greenwood/cli/src/types/index.d.ts
@@ -61,5 +61,5 @@ export type {
 export type CLI_COMMAND = "develop" | "build" | "serve";
 
 declare module "@greenwood/cli" {
-  export const run: (CLI_COMMAND) => Promise<void>;
+  export const run: (command: CLI_COMMAND) => Promise<void>;
 }
