diff --git a/node_modules/@greenwood/plugin-adapter-vercel/src/types/index.d.ts b/node_modules/@greenwood/plugin-adapter-vercel/src/types/index.d.ts
index abffd41..e21f89a 100644
--- a/node_modules/@greenwood/plugin-adapter-vercel/src/types/index.d.ts
+++ b/node_modules/@greenwood/plugin-adapter-vercel/src/types/index.d.ts
@@ -1,4 +1,4 @@
-import type { Plugin } from "@greenwood/cli";
+import type { AdapterPlugin } from "@greenwood/cli";
 
 type SUPPORTED_NODE_VERSIONS = "nodejs22.x" | "nodejs20.x" | "nodejs18.x";
 
@@ -6,7 +6,7 @@ type VercelAdapterOptions = {
   runtime?: SUPPORTED_NODE_VERSIONS;
 };
 
-export type VercelAdapter = (options?: VercelAdapterOptions) => Array<Plugin>;
+export type VercelAdapter = (options?: VercelAdapterOptions) => AdapterPlugin;
 
 declare module "@greenwood/plugin-adapter-vercel" {
   export const greenwoodPluginAdapterVercel: VercelAdapter;
